src/OssiaTypes.h

```diff
@@ -85,6 +86,51 @@ template<> struct MatchingType<double> {
   }
 };
 
+template<> struct MatchingType<ossia::impulse> {
+    using ofx_type = ossia::impulse;
+    static constexpr const auto val = ossia::val_type::IMPULSE;
+    using ossia_type = ossia::impulse;
+    using ossia_unit = ossia::unit_t;
+
+    static ofx_type convertFromOssia(const ossia::value& v)
+    {
+      return v.get<ossia_type>();
+    }
+
+    static ossia_type convert(ofx_type f)
+    {
+      return ossia::impulse();
+    }
+};
+
+template<> struct MatchingType<std::vector<string> > {
+  using ofx_type = std::vector<string>;
+  static constexpr const auto val = ossia::val_type::TUPLE;
+  using ossia_type = std::vector<ossia::value>;
+  using ossia_unit = ossia::unit_t;
+
+  static ofx_type convertFromOssia(const ossia::value& v)
+  {
+    const ossia_type& t = v.get<ossia_type>();
+    ofx_type _return{};
+    _return.reserve(t.size());
+    for_each(t.begin(), t.end(), [&_return](const ossia::value& value){
```

Comment:
also, if you insist on using explicit algorithms (which is generally a good initiative!) maybe a better suited algorithm would be std::transform.

It would look like : 

    std::transform(t.begin(), t.end(), std::back_inserter(_return), [] (const auto& value) { return value.get<std::string>(); });

There is a simplified version in `<ossia/detail/algorithms.hpp>` where the call would look like: 

    ossia::transform(t, std::back_inserter(_return), [] (const auto& value) { return value.get<std::string>(); });